src/components/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import UserIcon from 'images/icons/user.svg';
+import { useEffect, useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import { AppRoutes } from 'types/AppRoutes';
+
+import styles from './UserMenu.module.scss';
+import { UserMenuProps } from './UserMenu.types';
+
+export function UserMenu({ userName, className }: UserMenuProps) {
+  const [isActive, setActive] = useState(false);
+
+  useEffect(() => {
+    const documentClickHandler = () => setActive(false);
```

Comment:
❗️
Сейчас название ничего не говорит о том, что делает функция
```suggestion
    const toggleMenuActiveStatus = () => setActive(false);
```